client/mksubvolume.cc

```diff
@@ -326,45 +396,85 @@ doit()
     // Determine name for new subvolume: It is the target name without the
     // leading filesystem target.
 
-    string subvolume_name = target.substr(fs_target.size() + (fs_target == "/" ? 0 : 1));
+    string subvolume_name = target.substr(fs_target.size() +
+					 (fs_target == "/" || fs_target == target ? 0 : 1));
     if (verbose)
 	cout << "subvolume-name:" << subvolume_name << endl;
 
-    // Execute all steps.
+    // Set up cache for UUID / LABEL resolution in mnt_table_find_source
+    libmnt_cache* cache = mnt_new_cache();
```

Comment:
mnt_new_cache() could fail. More important is when cache gets freed, AFAIS a call to mnt_unref_cache() somethere later is missing.

Reply:
Error handling for `mnt_new_cache` is handled in `MntTable::set_cache` already. MntTable will now also dereference the cache automatically.